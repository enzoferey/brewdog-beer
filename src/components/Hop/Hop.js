import React, { Fragment } from "react";
import PropTypes from "prop-types";
import style from "./hop.scss";

import weigthIcon from "components/Table/weight.png";
import timeIcon from "components/Table/time.png";
import labelIcon from "components/Table/label.png";

import TableRowSection from "components/TableRowSection";
import IconLabel from "components/IconLabel";

const Hop = ({ name, amount: { value, unit }, add, attribute }) => (
  <Fragment>
    <TableRowSection>
      <span className={style.name}>{name}</span>
    </TableRowSection>
    <TableRowSection>
      <IconLabel icon={weigthIcon} label={`${value} ${unit}`} />
    </TableRowSection>
    <TableRowSection>
      <IconLabel icon={timeIcon} label={add} />
    </TableRowSection>
    <TableRowSection>
      <IconLabel icon={labelIcon} label={attribute} />
    </TableRowSection>
  </Fragment>
);

Hop.propTypes = {
  name: PropTypes.string,
  amount: PropTypes.shape({
    value: PropTypes.number,
    unit: PropTypes.string,
  }),
  add: PropTypes.oneOf(["start", "middle", "end"]),
  attribute: PropTypes.string,
};

export default Hop;
