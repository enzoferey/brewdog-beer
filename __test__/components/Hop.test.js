import React from "react";
import Hop from "components/Hop";

const hop = mocks.beers[0].ingredients.hops[0];

test("renders correctly", () => {
  testSnapshot(<Hop {...hop} />);
});
