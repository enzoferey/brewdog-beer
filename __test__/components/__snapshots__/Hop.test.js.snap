// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className={undefined}
    style={
        Object {
            "flex": 1,
          }
    }
>
    <span
        className={undefined}
    >
        Fuggles
    </span>
</div>,
  <div
    className={undefined}
    style={
        Object {
            "flex": 1,
          }
    }
>
    <img
        alt=""
        className={undefined}
        src="weight.png"
    />
    <span
        className={undefined}
    >
        25 grams
    </span>
</div>,
  <div
    className={undefined}
    style={
        Object {
            "flex": 1,
          }
    }
>
    <img
        alt=""
        className={undefined}
        src="time.png"
    />
    <span
        className={undefined}
    >
        start
    </span>
</div>,
  <div
    className={undefined}
    style={
        Object {
            "flex": 1,
          }
    }
>
    <img
        alt=""
        className={undefined}
        src="label.png"
    />
    <span
        className={undefined}
    >
        bitter
    </span>
</div>,
]
`;
